azureeyemodule/app/iot/iot_interface.cpp

```diff
@@ -215,14 +193,16 @@ static IOTHUBMESSAGE_DISPOSITION_RESULT receive_onvif_message(IOTHUB_MESSAGE_HAN
         }
     }
 
-    if (deserialized_msg.find(std::string("s")) != deserialized_msg.end())
+    if (json_object_get_value(root_object, "-s") != nullptr)
     {
-        const auto value = deserialized_msg[std::string("s")];
+        std::string value  = json_object_get_string(root_object, "-s");
         if (rtsp::is_valid_resolution(std::string(value)))
         {
             iot::update::restart_model_with_new_resolution(std::string(value));
-            rtsp::set_stream_params(rtsp::StreamType::RAW, std::string(value));
-            rtsp::set_stream_params(rtsp::StreamType::RESULT, std::string(value));
+            rtsp::set_stream_params(rtsp::StreamType::RAW, std::string(value), false);
```

Comment:
Seems like you are turning the streams off and then back on again. Is that intentional? It would force anyone currently viewing the stream to get the updated stream parameters, so it's fine, but if you don't mean to do that, the other way would be to do this:

```C++
iot::update::restart_model_with_new_resolution(std::string(value));
rtsp::set_stream_params(rtsp::StreamType::RAW, std::string(value));
rtsp::set_stream_params(rtsp::StreamType::RESULT, std::string(value));
```

that would not update any streams that are currently connected, but anytime someone connects in the future, they would get the new values. Up to you. But maybe add a comment explaining it?